lua/nvim-treesitter/locals.lua

```diff
@@ -24,6 +24,16 @@ function M.get_locals(bufnr)
   return queries.get_matches(bufnr, 'locals')
 end
 
+--- Creates unique id for a node based on text and range
+-- @param scope: the scope node of the definition
+-- @param bufnr: the buffer
+-- @param node_text: the node text to use
+-- @returns a string id
+function M.get_definition_id(scope, node_text)
+  -- Add a vaild starting character in case node text doesn't start with a valid one.
+  return table.concat({ 'k', node_text, scope:range() }, '_')
```

Comment:
why not node.id? Is node.id not unique when compared over time?

Reply:
Honestly I didn't even nodes had IDs. I don't think I saw it in the docs. I'm also not 100% sure if the node id will be the same across different trees...

Reply:
I guess the ID should change on reparsing. I am however not sure. You could  test it.